### PR TITLE
#Fix 2: Memoize Characters

### PR DESCRIPTION
**Pull request by ooade**
cc @olumytee @serkanatpaystack

### CODE REVIEW

**Comment by ooade** [comment]
Tho, as @serkanatpaystack mentioned earlier, this also could be an utility function.

**Comment by ooade** [comment]
Yeah, using context seem like a better approach to me too. I'll do that now, thanks.

File: src/contexts/Characters.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const fetchAllCharacters = async unresolvedCharacters => {
+    const characterPromise = unresolvedCharacters.map(character => {
+      return memoizedFetch(character);
+    });
+
+    const characters = await Promise.all(characterPromise);
+    setCharacters(characters);
+  };
+
+  const renderSortOrderArrow = type => {
```
Comment (serkanatpaystack) [inline src/contexts/Characters.js]: Why are you passing something visual like this in a data context?

Comment (serkanatpaystack) [reply src/contexts/Characters.js]: I'd suggest making it part of the character list ui and rendering based on the data/props

Comment (serkanatpaystack) [reply src/contexts/Characters.js]: this is conversational recommendation, you don't need to implement it.

**Comment by serkanatpaystack** [comment]
@ooade thank you, you can clean up, approve and merge this PR now, we don't need to iterate. I'll get back to you on another email. well done.

**Comment by ooade** [comment]
Yay! Thanks. I thought you wouldn't be at work now :sweat_smile: 

Thanks a lot.

**Comment by ooade** [comment]
Best feedbacks I've received in a very long time :tada: :tada:

**Comment by serkanatpaystack** [comment]
Huh yes, I did not want to end the day before replying you, you've worked hard, you deserve a reply. And thanks for appreciating the feedback. Enjoy your evening.

**Pull request merged**